SAMOA-65 Apache Kafka integration components for SAMOA

pwawrzyniak opened the pull request:
Initial version of Kafka components for SAMOA. Works with Apache Kafka 0.10, includes JSON and AVRO mappers to read/wtite instances to Kafka.

gdfm commented on samoa-api/src/main/java/org/apache/samoa/streams/kafka/KafkaEntranceProcessor.java:
```diff
@@ -0,0 +1,124 @@
+    @Override
+    public ContentEvent nextEvent() {
+        // assume this will never be called when buffer is empty!        
+        return this.deserializer.deserialize(buffer.remove(buffer.size() - 1));
```
Wouldn't this reverse the order of the messages?

pwawrzyniak replied:
Yes, I changed it so now the order would be correct (i.e. FIFO- items would be passed from this EntranceProcessor in the same order as they are received from Apache Kafka )

gdfm commented on samoa-api/src/main/java/org/apache/samoa/streams/kafka/KafkaTask.java:
```diff
@@ -0,0 +1,144 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.samoa.streams.kafka;
```
If this is supposed to be a stand alone task, it should probably go in org.apache.samoa.tasks

gdfm commented on samoa-api/src/test/java/org/apache/samoa/streams/kafka/topology/SimpleComponentFactory.java:
```diff
@@ -0,0 +1,53 @@
+package org.apache.samoa.streams.kafka.topology;
```
Where are these "Simple" classes used?
What is their purpose?

pwawrzyniak replied:
True- these classes are no longer used, therefore are removed now.

gdfm commented on .gitignore:
```diff
@@ -1,15 +1,16 @@
-.idea/
-*.iml
-*.iws
+#maven
```
There are a number of changes in the line endings, it seems to be a Linux/Windows problem.
Can you please revert them?

pwawrzyniak replied:
I had wrong setting in my git configuration. Should be fine now.

gdfm commented on samoa-api/pom.xml:
```diff
@@ -1,129 +1,154 @@
-    </plugins>
-  </build>
-</project>
+<?xml version="1.0" encoding="UTF-8"?>
```
There are a number of changes in the line endings, it seems to be a Linux/Windows problem.
Can you please revert them?

gdfm commented:
Hi,
@pwawrzyniak did you manage to reproduce the build problem?
Any idea what caused it and how to fix it?

gdfm commented on samoa-api/pom.xml:
```diff
@@ -11,119 +11,144 @@
   
```
I still see a number of whitespace changes.

pwawrzyniak commented:
Hi,
@gdfm  I reproduced an error and fixed that. The point is that samoa-samza uses Kafka libraries and there was version conflict (in particular the new API has different method signature compared to the version that was used in samoa-samza). I fixed the issue and in my environment all tests are passing now:
Reactor Summary:

Apache SAMOA ...................................... SUCCESS [4.246s]
samoa-instances ................................... SUCCESS [2.325s]
samoa-api ......................................... SUCCESS [49.973s]
samoa-test ........................................ SUCCESS [1.412s]
samoa-local ....................................... SUCCESS [54.005s]
samoa-threads ..................................... SUCCESS [2:00.199s]
samoa-storm ....................................... SUCCESS [1:03.299s]
samoa-apex ........................................ SUCCESS [11.334s]
samoa-flink ....................................... SUCCESS [13.020s]
samoa-samza ....................................... SUCCESS [12.428s]
------------------------------------------------------------------------
BUILD SUCCESS

gdfm commented:
+1, I think it is ready to be merged.

nicolas-kourtellis commented:
+1, lets merge this

gdfm commented:
I did a small change in the commit log to consolidate the changes in a single commit, as it was polluting the commit log.
As a result, I had to rewrite the history for the last patch.
Please force pull on your local master.
